clients/admin-ui/src/features/system/history/modal/fields/SystemCustomFieldGroup.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+
+import { useCustomFields } from "~/features/common/custom-fields/hooks";
+import { ResourceTypes } from "~/types/api";
+
+import SystemDataGroup from "../SystemDataGroup";
+import SystemDataTags from "./SystemDataTags";
+import SystemDataTextField from "./SystemDataTextField";
+
+interface SystemCustomFieldGroupProps {
+  customFields?: Record<string, any>;
+  resourceType: ResourceTypes;
+}
+
+const SystemCustomFieldGroup: React.FC<SystemCustomFieldGroupProps> = ({
+  customFields = {},
+  resourceType,
+}) => {
+  const { idToCustomFieldDefinition } = useCustomFields({
+    resourceType,
+  });
+
+  /** Used to determine if a custom field should be rendered as a text field or data tags.
+   * The presence of an allow_list_id indicates either a single or multi-value select
+   */
+  const isMultivalued = (name: string): boolean =>
+    Array.from(idToCustomFieldDefinition.values()).some(
+      (value) => value.name === name && !!value.allow_list_id
+    );
+
+  const prefix =
+    resourceType == ResourceTypes.SYSTEM
+      ? "custom_fields"
+      : "privacy_declarations[0].custom_fields";
+
+  // to ensure the order in the diff lists is the same
+  const sortedFieldNames = Object.keys(customFields).sort();
```

Comment:
This is the fix for the out-of-order fields, this will handle both the system info and privacy declaration fields.